### ISSUE TITLE
[NEWS] Zarządzanie czarną listą

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#1939]
Udostępniliśmy zasoby (w wersji beta), dzięki którym jako sprzedawca możesz [zarządzać czarną lista kupujących](https://allegro.pl/moje-allegro/sprzedaz/czarna-lista#/).
Osoba dodana do czarnej listy nie może kupować w twoich ofertach. W każdej chwili możesz usunąć z niej użytkownika, tym samym ponownie pozwolisz mu na zakup w twoich ofertach.

Aby dodać kupującego do czarnej listy, skorzystaj z zasobu[ POST /sale/blacklisted-users](https://developer.allegro.pl/documentation/#operation/doAddToBlackList).

**Ważne!** W strukturze musisz przesłać login lub id kupującego.

**Przykładowy request:**

```
curl -X POST \
https://api.allegro.pl/sale/blacklisted-users \
-H 'Authorization: Bearer {token}' \
-H 'Accept: application/vnd.allegro.beta.v1+json' \
-H 'Content-Type: application/vnd.allegro.beta.v1+json' \
-d ‘{
    "user": {
         "id": 123456,		          -- wymagane (jeśli nie przekazujesz loginu), 
                                          id kupującego
    	 "login": "bad_buyer"	          -- wymagane (jeśli nie przekazujesz id), 
                                          login kupującego
  	},
  	"note": "Rude person"             -- powód dodania do czarnej listy
}’
```
**Przykładowy response:**
```
{
    "user": {
        "id": 123456,
    	 "login": "bad_buyer"
  	},
    "note": "Rude person",
    "createdAt": "2019-05-08T09:45:818Z"   -- data dodania do czarnej listy
}
```

Aktualną czarną listę pobierzesz za pomocą zasobu
[GET /sale/blacklisted-users](https://developer.allegro.pl/documentation/#operation/doGetBlackListUsers)

**Przykładowy response:**

```
{
    "blacklistedUsers": [
        {
            "user": {
                "id": 123456,
                "login": "bad_buyer"
            },
            "note": "Rude person",
            "createdAt": "2019-07-04T10:41:31.135Z"
        },
        {
            "user": {
                "id": 012345,
                "login": "bad_buyer1"
            },
            "note": "Rude person",
            "createdAt": "2019-07-04T10:35:10.013Z"
        }
    ],
    "offset": 0,
    "limit": 25,			    -- limit wyświetlanych wyników przy
				            pojedynczym zapytaniu
    "total": 2			            -- liczba użytkowników przypisanych
				            do czarnej listy
}
```
Maksymalnie przy zapytaniu możesz wyświetlić do 25 wyników. Aby zwrócić kolejne, użyj parametru “offset”, gdzie jako wartość podaj wielokrotność limitu.

Aby usunąć kupującego z czarnej listy wykorzystaj zasób
[DELETE /sale/blacklisted-users/{userId}](https://developer.allegro.pl/documentation/#operation/doRemoveFromBlackList).

**Comment by allegrofan** [comment]
Po zmianach przestało działać. [error] => unauthorized [error_description] => Authorization header does not contain space after 'Bearer'

**Comment by PrzemyslawLukanowski** [comment]
@allegrofan Tutaj komunikat wskazuje raczej na inną przyczynę. Upewnij się, że w nagłówku Authorization wstawiasz spację między `Bearer` a tokenem.

**Comment by allegrofan** [comment]
Tak, komunikat wskazuje na inną przyczynę, problem w tym, że spacja jest i w innych wywołaniach działa bez problemu.

**Comment by PrzemyslawLukanowski** [comment]
Nie mogę potwierdzić, aby występował problem z działaniem zasobów - nasze logi wskazują, że wywołania metod POST i DELETE u innych użytkowników kończą się prawidłową odpowiedzią. Prześlij proszę przykładowy request wraz z response przez nasz [formularz kontaktowy](https://allegro.pl/pomoc/kontakt?kategoria=0cad5ed6-ca26-4669-8ff0-6a7da503fe80&subjectId=8b7d668e-6f52-47e4-8c6b-cfb5e985789a), zweryfikujemy to. W zgłoszeniu dopisz, że dane dotyczą wątku #1939 na GitHub.
